components/brave_rewards/browser/rewards_protocol_handler.cc

```diff
@@ -133,17 +62,71 @@ namespace brave_rewards {
 
 void HandleRewardsProtocol(const GURL& url,
                            content::WebContents::OnceGetter web_contents_getter,
-                           ui::PageTransition page_transition,
-                           bool has_user_gesture) {
-  DCHECK(url.SchemeIs(kRewardsScheme));
+                           ui::PageTransition page_transition) {
+  CHECK(url.SchemeIs(kRewardsScheme));
   content::GetUIThreadTaskRunner({})->PostTask(
       FROM_HERE,
       base::BindOnce(&LoadRewardsURL, url, std::move(web_contents_getter),
-                     page_transition, has_user_gesture));
+                     page_transition));
 }
 
 bool IsRewardsProtocol(const GURL& url) {
   return url.SchemeIs(kRewardsScheme);
 }
 
+absl::optional<GURL> ValidateWalletProviderRedirect(const GURL& referrer_url,
+                                                    const GURL& redirect_url) {
+  if (!referrer_url.is_valid() || !redirect_url.is_valid()) {
```

Comment:
can we also check that the referrer URLs have the expected origins/schemes?

Reply:
Scheme (`rewards://`) is already checked for the `redirect_url` (which is why the navigation is handled by the Rewards protocol handler). Did you want me to add
```cpp
|| !referrer_url.SchemeIsHTTPOrHTTPS()
```
to the conditions above?

Reply:
yes, if we expected referrer URL to always be HTTPS, it'd be good to add the second check